experimental/saiexperimentaldash.h

```diff
@@ -0,0 +1,2189 @@
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove inbound_acl_stage3
+ *
+ * @param[in] inbound_acl_stage3_id Entry id
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_remove_inbound_acl_stage3_fn)(
+        _In_ sai_object_id_t inbound_acl_stage3_id);
+
+/**
+ * @brief Set attribute for inbound_acl_stage3
+ *
+ * @param[in] inbound_acl_stage3_id Entry id
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_set_inbound_acl_stage3_attribute_fn)(
+        _In_ sai_object_id_t inbound_acl_stage3_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get attribute for inbound_acl_stage3
+ *
+ * @param[in] inbound_acl_stage3_id Entry id
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_get_inbound_acl_stage3_attribute_fn)(
+        _In_ sai_object_id_t inbound_acl_stage3_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Create eni_meter_entry
+ *
+ * @param[in] eni_meter_entry Entry
+ * @param[in] switch_id Switch id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_create_eni_meter_entry_fn)(
+        _In_ const sai_eni_meter_entry_t *eni_meter_entry,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove eni_meter_entry
+ *
+ * @param[in] eni_meter_entry Entry
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_remove_eni_meter_entry_fn)(
+        _In_ const sai_eni_meter_entry_t *eni_meter_entry);
+
+/**
+ * @brief Set attribute for eni_meter_entry
+ *
+ * @param[in] eni_meter_entry Entry
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_set_eni_meter_entry_attribute_fn)(
+        _In_ const sai_eni_meter_entry_t *eni_meter_entry,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get attribute for eni_meter_entry
+ *
+ * @param[in] eni_meter_entry Entry
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_get_eni_meter_entry_attribute_fn)(
+        _In_ const sai_eni_meter_entry_t *eni_meter_entry,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+typedef struct _sai_dash_api_t
```

Comment:
In order to support atomic operation we need to get list of ACLs.
Need to extend the api to allow to remove an acl lists, add acl lists in the same api. 
In the sirius api - updating of the ACL Group (ACLStage) must be an atomic operation. No partial updates are allowed, as it might lead to security issues in the case of only partial rules being applied.

Reply:
It will be also much more efficient to allow to set list of eni / routes  and all the other configuration in a list and not one by one.
If we are talking about ca2pa for example , if we want to insert few thousands rules, it will be much better to get them in a list and not one by one